### DISCUSSION TITLE
Excessive number of guests go to a new ride in parks with large numbers of guests

### DISCUSSION THREAD

**Discussion by CSEverett** [OpenRCT2/OpenRCT2#18111] [Feature requests]
I've been playing two older parks with around 7,000 guests each, and when a new ride is opened, you can easily get 400-600+ guests heading for it. Which proved a problem when one could only handle around 1200 guests per hour, and can easily lead to guest overcrowding issues and a few hundred leaving the park. And you have to plan for the massive influx of guests to that part of the park. And _really_ hope it doesn't rain. I'm not sure what can be done about it, if anything, but it is an issue.

**Comment by ocalhoun6** [comment]
Maybe when a guest is deciding which ride to head for, they could check if the queue is full, and then eliminate that one from the list of possibilities if it's full? (The same way a peep will avoid heading toward a ride that's currently closed.) Then they'll pick a different ride instead. That wouldn't entirely solve this problem, but it might help. Lots of peeps might still decide to head for a very new ride that hasn't filled the queue yet, but once the queue is full, the number of peeps choosing to head for the ride should decrease to about what the ride's throughput can handle.

-----

Since new rides apparently are more weighted when a peep is deciding which ride to head for, maybe that weight factor could be changed so that it's divided by the number of rides in the park. So a park with few rides will have peeps flocking to new rides, but in a park with many rides, new rides will get very little extra attention.

That would mostly fix the problem. Because in order to get that many guests (without using cheats or abusing game mechanics) you'd have to have quite a lot of rides. And the formula could be tweaked to ensure it has the same behavior as vanilla for, say, up to 50 rides.

You could also address the problem more directly by dividing this weight factor by the number of guests in the park.

**Reply by CSEverett** [reply]
Ok this is getting ridiculous. Just opened a new section of the park. For reference, "The Grand" refers to the coaster on the top with the solid line of guests heading for it.  You kind of have to use cheats to enhance capacity. (And there's already 109 in the queue):
![Queens Island Modern 2022-09-26 18-55-41](https://user-images.githubusercontent.com/89952921/192403085-2ef32d90-9020-4ece-8a7f-a510b0f14388.png)

**Comment by astraylife1** [comment]
I have had this occur as well. I have parks with 7000-10000 guests and whenever a new ride is open, the peeps will flood it for at least a year or two of in-game time. It is ridiculous.

**Reply by CSEverett** [reply]
Agreed. You almost need to be using the disable vehicle limits cheat for longer trains in order to have enough capacity to handle it. When I built a flying turns/wooden bobsled coaster that "only" does ~1250 guests per hour, it created a huge mess of crowds swarming it. Excessively long queue are almost a most to avoid losing hundreds of guests from overcrowding, as is the ability to have 100+ guests on the ride at once.

**Reply by astraylife1** [reply]
The problem with creating a giant queue or giving it huge capacity is that in two or so game years, the crowds will cease and things will settle down to normal. And then you’re left with a ride they no longer needs the capacity and you’ve got an empty queue.

**Reply by CSEverett** [reply]
Tell me about it. (Sigh.) Clever ride siting and footpath design can help make up for it, but only so much. And extending the queue entrance partway into a multi-tile wide (separated by fences) fairway doesn't look good at all. My vertical drop coaster wound up with (by accident) massive capacity, and it's at the end of a footpath branch so guests have to turn around or walk right into the queue.  Depending on where in the park the guests have gravitated to it can run at a fair portion of capacity, but if that arm of the fairway gets extended it'll never run full again. And you have to close a ride and get everybody off to reduce the number of trains, but if you do that you won't be able to fill the remaining ones. Although every few years you to have to (or should) close a ride for refurbishment once it drops below 50% condition., so you can do it then. Multiple rides built nearby in a short period of time used to be crammed to the brim, and now run mostly empty trains with a year or two having passed and new rides open elsewhere.

**Reply by astraylife1** [reply]
Completely understand. Sometimes I try to open two new rides at once, but they are located on opposite sides of the park to try and separate the crowds a bit. Still causes a mess

**Reply by CSEverett** [reply]
You really do need at least triple wide (separated by fences) footpaths in many areas, and sometimes 4-wide. I mean, in the screenshot above, 200-250 guests heading for The Grand would be fine and about right, but not 700 plus! And hope it doesn't rain.

**Comment by CSEverett** [comment]
Actually, I think the best thing you could do it provide some sort of weighting based on the number of seats in a ride (peeps per car x number of cars per train x number of trains), which could serve as a proxy for capacity so huge high-capacity new rides get a deluge, but smaller lower-capacity rides get fewer. That would at least help reduce overcrowding/more guests then the ride could handle, while also allowing rides that actually can adsorb 500 guests coming their way will get them.

**Comment by astraylife1** [comment]
I tried an experiment. I built a new coaster and set the price so high that no guest would ride. I left it open for 3 years, running empty since everyone said it was too expensive. Then I dropped the price after 3 years and still, half the park came flooding to this coaster that should no longer be considered new. And with a park of 10,000 guests, when all was said and done, 3,000 guests had left due to how crowded the area was by the "new" coaster. Something is up with the game mechanics that doesn't really consider age of the ride in this instance.

**Comment by J-PIE-314** [comment]
This is another request to solve this issue: #15825.
